packages/backend-core/src/util/errors.ts

```diff
@@ -1,13 +1,60 @@
-export class JWTExpiredError extends Error {
-  constructor(message: string) {
-    super(message);
-    this.name = 'JWTExpiredError';
+import { AuthErrorReason, TokenType, TokenVerificationErrorReason } from '../types';
+
+/* All expected errors thrown calling base.verifySessionToken should be of this type */
+export class TokenVerificationError extends Error {
+  reason: TokenVerificationErrorReason;
+  constructor(reason: TokenVerificationErrorReason) {
+    super();
+    this.reason = reason;
   }
 }
 
-export class MissingJWTVerificationKeyError extends Error {
-  constructor(message: string) {
-    super(message);
-    this.name = 'MissingJWTVerificationKeyError';
+/**
+ * The JWT verification process does not need to know about the type of token (header, cookie etc.)
+ * To translate the generic errors to mapped AuthErrorReason header responses we map them based on the token type.
+ *
+ * Here we also steer the flow based on the verification error.
+ * `shouldSignout` set as true will lead to a signout, otherwise we throw the interstitial
+ *
+ * TODO: Can be probably done better with template literals
```

Comment:
FYI due to the late hours, my mind was not in capable state to add some smarts to reduct this switch.

Reply:
lgtm - sometimes the unreduced is more readable